tests/migration/nodeselector.go

```diff
@@ -0,0 +1,247 @@
+		Expect(err).ToNot(HaveOccurred())
+		vmi = libwait.WaitForSuccessfulVMIStart(vmi, libwait.WithTimeout(180))
+
+		sourceNodeName := vmi.Status.NodeName
+		var targetNodeName string
+
+		By("labeling a target node")
+		for _, node := range nodes.Items {
+			if node.Name != sourceNodeName {
+				targetNodeName = node.Name
+				libnode.AddLabelToNode(node.Name, cleanup.TestLabelForNamespace(vmi.Namespace), "target")
+				break
+			}
+		}
+		Expect(targetNodeName).ToNot(BeEmpty(), "There should be a labeled target node")
+
+		By("Checking nodeSelector on the VMI")
+		Expect(vmi.Spec.NodeSelector).ToNot(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+
+		By("Checking nodeSelector on virt-launcher pod")
+		virtLauncherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(virtLauncherPod.Spec.NodeSelector).ToNot(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+
+		By("Starting the migration to the labeled node")
+		migration := libmigration.New(vmi.Name, vmi.Namespace)
+		migration.Spec.AddedNodeSelector = map[string]string{cleanup.TestLabelForNamespace(vmi.Namespace): "target"}
+		libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(virtClient, migration)
+
+		By("Checking that the VMI landed on the target node")
+		vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.GetName(), metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(vmi.Status.NodeName).To(Equal(targetNodeName))
+
+		By("Checking nodeSelector on the VMI")
+		Expect(vmi.Spec.NodeSelector).ToNot(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+
+		By("Checking nodeSelector on virt-launcher pod")
+		virtLauncherPod, err = libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(virtLauncherPod.Spec.NodeSelector).To(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+
+		By("Migrating again the VM without configuring a nodeselector")
+		migration = libmigration.New(vmi.Name, vmi.Namespace)
+		libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(virtClient, migration)
+
+		By("Checking nodeSelector on the VMI")
+		vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.GetName(), metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(vmi.Spec.NodeSelector).ToNot(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+
+		By("Checking nodeSelector on virt-launcher pod")
+		virtLauncherPod, err = libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(virtLauncherPod.Spec.NodeSelector).ToNot(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+	})
+
+	It("Should fail the migration when the nodeSelector could not be satisfied", decorators.Periodic, func() {
```

Comment:
I believe this is already covered "should automatically cancel unschedulable migration after a timeout period"

Reply:
Maybe you want to convert the part where you check that VMI is not touched to unit test.

Reply:
done, done